aodntools/timeseries_products/velocity_aggregated_timeseries.py

```diff
@@ -0,0 +1,316 @@
+            ## clip to in water data only
+            nc = in_water(nc)
+
+            varlen_file.append(get_nvalues(nc))
+            error_list = check_file(nc, site_code)
+            if not error_list:
+                varlen_list.append(get_nvalues(nc)[0])
+            else:
+                bad_files.append([file, error_list])
+                rejected_files.append(file)
+
+    ## remove bad files form the list
+    for file in bad_files:
+        files_to_agg.remove(file[0])
+
+
+    varlen_list = [0] + varlen_list
+    varlen_total = sum(varlen_list)
+    n_files = len(files_to_agg)
+
+
+    ## create ncdf file, dimensions and variables
+    ds = nc4.Dataset(os.path.join(base_path, outfile), 'w')
+    OBSERVATION = ds.createDimension('OBSERVATION', size=varlen_total)
+    INSTRUMENT = ds.createDimension('INSTRUMENT', size=n_files)
+
+    UCUR = ds.createVariable(varname='UCUR', datatype='float', zlib=True, dimensions=('OBSERVATION'), fill_value=999999.0)
+    VCUR = ds.createVariable(varname='VCUR', datatype='float', zlib=True, dimensions=('OBSERVATION'), fill_value=999999.0)
+    WCUR = ds.createVariable(varname='WCUR', datatype='float', zlib=True, dimensions=('OBSERVATION'), fill_value=999999.0)
+    DEPTH = ds.createVariable(varname='DEPTH', datatype='float', zlib=True, dimensions=('OBSERVATION'), fill_value=999999.0)
+    UCURqc = ds.createVariable(varname='UCUR_quality_control', datatype='byte', zlib=True, dimensions=('OBSERVATION'), fill_value=99)
+    VCURqc = ds.createVariable(varname='VCUR_quality_control', datatype='byte', zlib=True, dimensions=('OBSERVATION'), fill_value=99)
+    WCURqc = ds.createVariable(varname='WCUR_quality_control', datatype='byte', zlib=True, dimensions=('OBSERVATION'), fill_value=99)
+    DEPTHqc = ds.createVariable(varname='DEPTH_quality_control', datatype='byte', zlib=True, dimensions=('OBSERVATION'), fill_value=99)
+    TIME = ds.createVariable(varname='TIME', datatype='double', zlib=True, dimensions=('OBSERVATION'))
+    instrument_index = ds.createVariable(varname='instrument_index', datatype='int', zlib=True, dimensions=('OBSERVATION'))
+
+    source_file = ds.createVariable(varname='source_file', datatype='S256', dimensions=('INSTRUMENT'))
+    instrument_id = ds.createVariable(varname='instrument_id', datatype='S256', dimensions=('INSTRUMENT'))
+    LATITUDE = ds.createVariable(varname='LATITUDE', datatype='float', dimensions=("INSTRUMENT"))
+    LONGITUDE = ds.createVariable(varname='LONGITUDE', datatype='float', dimensions=("INSTRUMENT"))
+    NOMINAL_DEPTH = ds.createVariable(varname='NOMINAL_DEPTH', datatype='float', dimensions=('INSTRUMENT'))
+
+    ## main loop
+    for index, file in enumerate(files_to_agg):
+        print(index)
+        with xr.open_dataset(file) as nc:
+
+            ## in water data only
+            nc = in_water(nc)
+
+            start = sum(varlen_list[:index + 1])
+            end = sum(varlen_list[:index + 2])
+            n_cells = get_nvalues(nc)[1]
```

Comment:
Again - it's not easy to guess what the the second argument returned by get_nvalues is because the function name is not useful (you have to dig into the docstring).

Reply:
solved